Source/WebCore/platform/graphics/texmap/TextureMapperLayer3DRenderingContext.h

```diff
@@ -44,38 +46,45 @@ class TextureMapperLayer3DRenderingContext final {
         const std::function<void(TextureMapperLayer*, const ClipPath&)>&);
 
 private:
-    enum class PolygonPosition {
+    enum class LayerPosition {
         InFront,
         Behind,
         Coplanar,
         Intersecting
     };
 
-    struct TextureMapperLayerPolygon final {
+    struct Layer final {
         FloatPolygon3D geometry;
-        TextureMapperLayer* layer = { nullptr };
-        bool isSplitted = { false };
+        TextureMapperLayer* textureMapperLayer { nullptr };
+        bool isSplitted { false };
         unsigned clipVertexBufferOffset { 0 };
     };
 
-    struct TextureMapperLayerNode final {
+    struct LayerNode final {
         WTF_MAKE_STRUCT_FAST_ALLOCATED;
 
-        explicit TextureMapperLayerNode(TextureMapperLayerPolygon&& polygon)
+        explicit LayerNode(Layer&& layer)
         {
-            polygons.append(WTFMove(polygon));
+            layers.append(WTFMove(layer));
         }
 
-        const TextureMapperLayerPolygon& firstPolygon() const  { return polygons[0]; }
+        const Layer& firstLayer() const  { return layers[0]; }
 
-        Vector<TextureMapperLayerPolygon> polygons;
-        std::unique_ptr<TextureMapperLayerNode> frontNode;
-        std::unique_ptr<TextureMapperLayerNode> backNode;
+        Vector<Layer> layers;
+        std::unique_ptr<LayerNode> frontNode;
+        std::unique_ptr<LayerNode> backNode;
     };
 
-    void buildTree(TextureMapperLayerNode&, Deque<TextureMapperLayerPolygon>&);
-    void traverseTree(TextureMapperLayerNode&, const std::function<void(TextureMapperLayerNode&)>&);
-    static PolygonPosition classifyPolygon(const TextureMapperLayerPolygon&, const FloatPlane3D&);
+    using SweepAndPrunePairs = HashSet<std::pair<size_t, size_t>>;
+
+    static SweepAndPrunePairs sweepAndPruneXAxis(const Vector<Layer>&);
+    static FloatQuad projectLayerToZYPlane(const Layer&);
```

Comment:
I'd keep all those helpers inlined in the .cpp file, instead of declaring them here.

Reply:
Moved all functions that don't use private structs to .cpp static inlines